phpBB/adm/style/tooltip.js

```diff
@@ -10,206 +10,173 @@ phpBB Development Team:
 	}
 
-	for (i = 0; i < links.length; i++) {
-		if (sub_id) {
-			if (links[i].parentNode.id.substr(0, sub_id.length) === sub_id) {
-				prepare(links[i]);
+	$links.each(function () {
+		var $this = $(this);
+
+		if (subId) {
+			if ($this.parent().attr('id').substr(0, subId.length) === subId) {
+				phpbb.prepareTooltips($this, headline);
 			}
 		} else {
-			prepare(links[i]);
+			phpbb.prepareTooltips($this, headline);
 		}
-	}
-
-	tooltip_mode = 'select';
-}
+	});
+};
 
 /**
-* Prepare elements to replace
+ * Prepare elements to replace
+ *
+ * @param {object} $element Element to prepare for tooltips
+ * @param {string} headText Text heading to display
 */
-function prepare(element) {
+phpbb.prepareTooltips = function ($element, headText) {
 	var tooltip, text, desc, title;
 
-	text = element.getAttribute('title');
+	text = $element.attr('data-title');
 
 	if (text === null || text.length === 0) {
 		return;
 	}
 
-	element.removeAttribute('title');
-	tooltip = create_element('span', 'tooltip');
-
-	title = create_element('span', 'top');
-	title.appendChild(document.createTextNode(head_text));
-	tooltip.appendChild(title);
+	title = phpbb.createElement('span', 'top');
+	title.appendChild(document.createTextNode(headText));
 
-	desc = create_element('span', 'bottom');
+	desc = phpbb.createElement('span', 'bottom');
 	desc.innerHTML = text;
-	tooltip.appendChild(desc);
 
-	set_opacity(tooltip);
+	tooltip = phpbb.createElement('span', 'tooltip');
+	tooltip.appendChild(title);
+	tooltip.appendChild(desc);
 
-	element.tooltip = tooltip;
-	element.onmouseover = show_tooltip;
-	element.onmouseout = hide_tooltip;
-
-	if (tooltip_mode === 'link') {
-		element.onmousemove = locate;
-	}
-}
+	tooltips[$element.attr('data-id')] = tooltip;
+	$element.on('mouseover', phpbb.showTooltip);
+	$element.on('mouseout', phpbb.hideTooltip);
+};
 
 /**
-* Show tooltip
+ * Show tooltip
+ *
+ * @param {object} $element Element passed by .on()
 */
-function show_tooltip(e) {
-	document.getElementById('_tooltip_container').appendChild(this.tooltip);
-	locate(this);
-}
+phpbb.showTooltip = function ($element) {
+	var $this = $($element.target);
+	$('#_tooltip_container').append(tooltips[$this.attr('data-id')]);
+	phpbb.positionTooltip($this);
+};
 
 /**
-* Hide tooltip
+ * Hide tooltip
 */
-function hide_tooltip(e) {
+phpbb.hideTooltip = function () {
 	var d = document.getElementById('_tooltip_container');
 	if (d.childNodes.length > 0) {
 		d.removeChild(d.firstChild);
 	}
-}
-
-/**
-* Set opacity on tooltip element
-*/
-function set_opacity(element) {
-	element.style.filter = 'alpha(opacity:95)';
-	element.style.KHTMLOpacity = '0.95';
-	element.style.MozOpacity = '0.95';
-	element.style.opacity = '0.95';
-}
+};
 
 /**
-* Create new element
+ * Create new element
+ *
+ * @param {string} tag HTML tag
+ * @param {string} c Element's class
+ *
+ * @return {object} Created element
 */
-function create_element(tag, c) {
+phpbb.createElement = function (tag, c) {
```

Comment:
By 2.608 nanoseconds! http://jsperf.com/jquery-create-vs-native-element

That's not a difference worth caring about :P

http://blog.methvin.com/2013/12/please-stop-jsperfcom-abuse.html